File: scraper/src/app/models/licitacao.py
import uuid
from tortoise import fields
from tortoise.models import Model

from app.models.licitacao_item import LicitacaoItem


class Licitacao(Model):
    id = fields.UUIDField(pk=True, default=uuid.uuid4)
    unidade = fields.TextField(null=True)
    uasg = fields.IntField()
    numero_pregao = fields.IntField()
    objeto = fields.TextField(null=True)
    horario_edital = fields.CharField(max_length=255, null=True)
    endereco = fields.CharField(max_length=255, null=True)
    telefone = fields.CharField(max_length=255, null=True)
    fax = fields.CharField(max_length=255, null=True)
    entrega_da_proposta = fields.DatetimeField(null=True)
    itens: fields.ReverseRelation["LicitacaoItem"]
    
    class Meta:
        table = "licitacao"
        unique_together = (("uasg", "numero_pregao"),)

File: backend/src/main/java/com/example/demo/models/entities/Licitacao.java
package com.example.demo.models.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;
import java.util.UUID;

@Table("licitacao")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Licitacao {

    @Id
    private UUID id;
    private String unidade;
    private Integer uasg;
    private Integer numeroPregao;
    private String objeto;
    private String horarioEdital;
    private String endereco;
    private String telefone;
    private String fax;
    private LocalDateTime entregaDaProposta;

}
